wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt

```diff
@@ -2235,8 +2330,13 @@ class KotlinGenerator private constructor(
       return when (encodeMode!!) {
         EncodeMode.MAP ->
           Map::class.asTypeName().parameterizedBy(keyType.typeName, valueType.typeName)
-        EncodeMode.REPEATED,
-        EncodeMode.PACKED -> List::class.asClassName().parameterizedBy(type.typeName)
+        EncodeMode.PACKED -> {
+          when {
+            useArray -> primitiveArrayClassForType.asTypeName()
```

Comment:
I think there’s some places where we check `useArray` without checking encodeMode? I think we probably gotta assert elsewhere that you can’t have `useArray` without PACKED, otherwise I think we might generate an inconsistent file?